user-fund-storage/src/user_data.rs

```diff
@@ -33,4 +40,24 @@ pub trait UserDataModule {
         self.set_num_users(num_users);
         num_users
     }
+
+    fn get_or_create_user(&self, address: &Address) -> usize{
+        let mut user_id = self.get_user_id(&address);
+        if user_id == 0 {
+            user_id = self.new_user();
+            self.set_user_id(&address, user_id);
+            self.set_user_address(user_id, &address);
```

Comment:
you need something more here. even is there is a user_id you might need to set_user_address is that does not exist. The problem will be when some already delegated user will stake again and enters the waiting list. From that moment onwards waiting list view function will panic.